src/utils/magitUtils.ts

```diff
@@ -95,6 +97,16 @@ export default class MagitUtils {
     return window.showQuickPick(refs, { placeHolder: prompt });
   }
 
+  public static async chooseCommit(repository: MagitRepository, prompt: string) {
+
+    const log =
+      repository.log({ maxEntries: 100 })
+        .then(log => log.map(commit => { return { label: commit.hash, description: commit.message }; }));
+
+    // return window.showQuickPick(commits, { placeHolder: prompt })
+    return window.showQuickPick(log, { placeHolder: prompt, matchOnDescription: true }).then(item => item ? item.label : undefined);
+  }
+
```

Comment:
This is great. I agree that quick pick style is great. And it makes sense to use a VSCode interface, using its strengths, instead of emulating Emacs completely.